lua/neoclip/sorted_set.lua

```diff
@@ -47,6 +47,21 @@ local insert = function(self, entry, opts)
     end
 end
 
+local replace = function(self, entry, newEntry)
+    local key = hash(entry)
+    local new_key = hash(newEntry)
+    local node = self.entries[key];
```

Comment:
should the case when the `entry` is not already in the set be handled somehow?

Reply:
I think it's a good idea, but no other function has error handling. so I'm not sure how I'd got about that. I think I'll just call `error` with a good error message. If you have a better suggestion, let me know!

Reply:
I so far used [this warn function] for errors. Do you know if either has advantage? I haven't used `error` before?

Reply:
I see. the `error` function is a little different, in that it will guarantee a stop in the current stack execution. It acts like a `throw` in another languages (the equivalent `catch` would be a `pcall`). This can be both good and bad. [Here's](https://github.com/AckslD/nvim-neoclip.lua/blob/main/lua/neoclip/db.lua#L8) an example i which stopping the current lua execution would be really bad. But I think sometimes it can be good to make sure the user can't do anything else if the current function fails.

In this case, I think it makes sense to throw a proper error, indicating that the function failed to do it's job. This may prevent the user from assuming the entry has been edited when it has not. What are your thoughts on that?